### PR TITLE
GPII-2110: Added support for JSON 5

### PR DESCRIPTION
**Pull request by kaspermarkus**

### CODE REVIEW

**Comment by amb26** [comment]
Looks brilliant! Could you also fix up config loading whilst you are here?
https://github.com/fluid-project/kettle/blob/master/lib/KettleConfigLoader.js#L104 (factor this into a function which checks the file extension and then chooses which decoder to use) - Cheers!

**Comment by colinbdclark** [comment]
This is a fantastic addition to Kettle, @kaspermarkus. Nice work!

**Comment by kaspermarkus** [comment]
@amb26 done

File: lib/KettleUtils.js [hunk 1]
```diff
@@ -115,21 +116,31 @@ kettle.syncFilePromise = function (fileName, message) {
...
 };
 
-fluid.registerNamespace("kettle.JSON");
+/**
+ * Parses a config file by chosing the appropriate decoder based on the extension of the file
+ * Currently json and json5 are supported
+ *
+ * @return {Object} The parser to use (defaulting to json)
+ */
+kettle.getSuitableParser = function (fileName) {
```
Comment (amb26) [inline lib/KettleUtils.js]: I think sticking these both in a kettle.JSON namespace is still fine

File: lib/KettleConfigLoader.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ kettle.config.createDefaultsImpl = function (prefix, filePath) {
         fullPath = path.resolve(prefix, fileName);
     }
     var configFile;
-    var parser = kettle.JSON.readFileSync(fullPath, "reading config file at " + fullPath);
+    var parser = kettle.readFileSync(fullPath, "reading config file at " + fullPath);
```
Comment (amb26) [inline lib/KettleConfigLoader.js]: Please add a test using a JSON5-formatted config file (converting one of the existing ones to JSON5 is fine).
Please also add relevant explanation to docs/ConfigsAndApplictions.md - that is, at relevant times where it says "JSON" write "JSON or JSON5" together with a link to http://json5.org/

**Comment by kaspermarkus** [comment]
@amb26 done

**Pull request merged**